core/clustersmngr/clustersmngr.go

```diff
@@ -92,20 +98,27 @@ func ClientConfigWithUser(user *auth.UserPrincipal) ClusterClientConfig {
 			}
 		}
 
+		// flowcontrol.IsEnabled makes a request to the K8s API of the cluster stored in the config.
+		// It does a HEAD request to /livez/ping which uses the config.Dial timeout. We can use this
+		// function to error early rather than wait to call client.New.
 		enabled, err := flowcontrol.IsEnabled(context.Background(), config)
```

Comment:
Just needed to double check this but:

From https://pkg.go.dev/net/http
> A non-2xx status code doesn't cause an error.

So the behaviour here shouldn't change even for < 1.23 clusters 👍 .

Reply:
> A 404 to livez/ping (added in k8s 1.23) will not cause an error from flowcontrol.IsEnabled.

🤦 . I need to read better.